File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/EventDetailReportDto.java
package org.example.bailaconmigo.DTOs;

import lombok.Data;
import org.example.bailaconmigo.Entities.Enum.DanceStyle;
import org.example.bailaconmigo.Entities.Enum.EventStatus;
import org.example.bailaconmigo.Entities.Enum.EventType;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class EventDetailReportDto {
    private Long eventId;
    private String eventName;
    private EventType eventType;
    private EventStatus eventStatus;
    private LocalDateTime eventDate;
    private String organizerName;
    private String cityName;
    private String countryName;

    // Capacidad y ocupación
    private Integer capacity;
    private int totalRegistrations;
    private int confirmedRegistrations;
    private int pendingRegistrations;
    private int cancelledRegistrations;
    private int availableCapacity;
    private double occupancyRate;

    // Información del evento
    private Double price;
    private List<DanceStyle> danceStyles;

    // Métricas financieras
    private BigDecimal totalRevenue;

    // Calificaciones
    private double averageRating;
    private int totalRatings;

    // Participantes
    private List<ParticipantInfoDto> participants;
}

File: frontend/src/app/models/event-detail-report-dto.ts
import { EventType } from "./event-type";
import { EventStatus } from "./EventStatus";
import { ParticipantInfoDto } from "./participant-info-dto";

export interface EventDetailReportDto {
  eventId: number;
  eventName: string;
  eventType: EventType;
  eventStatus: EventStatus;
  eventDate: string;
  organizerName: string;
  cityName: string;
  countryName: string;
  capacity: number;
  totalRegistrations: number;
  confirmedRegistrations: number;
  pendingRegistrations: number;
  cancelledRegistrations: number;
  availableCapacity: number;
  occupancyRate: number;
  price: number;
  danceStyles: String[];
  totalRevenue: number;
  averageRating: number;
  totalRatings: number;
  participants: ParticipantInfoDto[];
}
